tests/e2e/testsLibrary/WorkspaceHandlingTests.ts

```diff
@@ -15,46 +15,95 @@ import { Dashboard } from '../pageobjects/dashboard/Dashboard';
+    }
+
     constructor(
         @inject(CLASSES.Dashboard) private readonly dashboard: Dashboard,
         @inject(CLASSES.CreateWorkspace) private readonly createWorkspace: CreateWorkspace,
         @inject(CLASSES.Workspaces) private readonly workspaces: Workspaces,
-        @inject(CLASSES.BrowserTabsUtil) private readonly browserTabsUtil: BrowserTabsUtil) {}
+        @inject(CLASSES.DriverHelper) private readonly driverHelper: DriverHelper,
+        @inject(CLASSES.BrowserTabsUtil) private readonly browserTabsUtil: BrowserTabsUtil,
+        @inject(CLASSES.Editor) private readonly editor: Editor,
+        @inject(CLASSES.Ide) private readonly ide: Ide,
+        @inject(CLASSES.ProjectAndFileTests) private readonly projectAndFileTests: ProjectAndFileTests) {}
 
     public createAndOpenWorkspace(stack: string) {
-        test(`Open 'New Workspace' page`, async () => {
+        test(`Create and open new workspace, stack:${stack}`, async () => {
             await this.dashboard.waitPage();
             await this.dashboard.clickCreateWorkspaceButton();
             await this.createWorkspace.waitPage();
-            const parentGUID = await this.browserTabsUtil.getCurrentWindowHandle();
+            WorkspaceHandlingTests.parentGUID = await this.browserTabsUtil.getCurrentWindowHandle();
             await this.createWorkspace.clickOnSample(stack);
-            await this.browserTabsUtil.switchToWindow(parentGUID);
-            WorkspaceHandlingTests.workspaceName = await this.dashboard.getRecentWorkspaceName(10000);
-            Logger.debug(`Workspace Name is: ${WorkspaceHandlingTests.workspaceName}`);
-            await this.browserTabsUtil.waitAndSwitchToAnotherWindow(parentGUID, 10000);
+            await this.browserTabsUtil.waitAndSwitchToAnotherWindow(WorkspaceHandlingTests.parentGUID, TimeoutConstants.TS_IDE_LOAD_TIMEOUT);
         });
     }
 
     public openExistingWorkspace(workspaceName: string) {
-        test('Start workspace', async () => {
+        test('Open and start existing workspace', async () => {
             await this.dashboard.waitPage();
             await this.dashboard.clickWorkspacesButton();
             await this.workspaces.waitPage();
             await this.workspaces.clickOpenButton(workspaceName);
         });
     }
 
+    public obtainWorkspaceNameFromApplicationYaml(yamlPath: string, fileName: string, editorLine: number) {
+        this.projectAndFileTests.openFile(yamlPath, fileName);
+        test(`Obtain workspace name from ${yamlPath}/${fileName}`, async() => {
+            let workspaceNameLine: string = await this.editor.getLineText(fileName, editorLine);
+            Logger.info(`Obtained name from application.yaml: ${workspaceNameLine}`);
+            WorkspaceHandlingTests.workspaceName = workspaceNameLine.split(`: `)[1];
+            CheReporter.registerRunningWorkspace(WorkspaceHandlingTests.workspaceName);
+        });
+    }
+
+    public switchBackToIdeTab() {
```

Comment:
it's actually switching to the FIRST tab from the left with IDE inside

Reply:
done.

Reply:
Thank you!
It's needed to fix method name in logs as well, isn't it?
```
test('WorkspaceHandlingTests.switchBackToIdeTab ...
```